src/libraries/System.Threading.Thread/tests/ThreadTests.cs

```diff
@@ -1090,7 +1116,24 @@ public static void StartTest()
                     Assert.Same(t, Thread.CurrentThread);
                 });
             t.IsBackground = true;
-            t.Start(t);
+            StartParameter(t, t);
+            waitForThread();
+
+            var al = new AsyncLocal<int>();
+            al.Value = 42;
+            t = ThreadTestHelpers.CreateGuardedThread(out waitForThread, parameter =>
+            {
+                if (useUnsafeStart)
+                {
+                    Assert.Equal(0, al.Value);
```

Comment:
could also check that culture didn't flow?

Reply:
Why?

Reply:
Just something else that wouldn't flow. But I guess this is fine.

Reply:
It flows via AsyncLocals.

Reply:
Ah I didn't know that.

Reply:
Yeah, in .NET Core, ExecutionContext is entirely about AsyncLocal... for anything to flow, it's done via AsyncLocal.